### ISSUE TITLE
We need to internally identify a file as BETA in the comment

### ISSUE THREAD

**Issue by asmusf** [unicode-org/unicodetools#813]
Now that we no longer have suffixes for property file drafts it is essential that we identify non-final files **internally**, possibly with something as simple as the word "BETA" in the first line.

Or something like "This is a BETA REVIEW DRAFT" on a line by itself.

The reason I'm adding this issue here, is that it may be possible to put this boilerplate into tooling. 

Now, the platinum solution would be a text block that also puts the link to the BETA PRI in each file, so something like 

> This is a BETA REVIEW DRAFT. For details on the beta review process and how to submit feedback see PRI nnn \<link\>.

That way, anyone receiving a copy of this file can immediately tell that
- this is not a final public version of the data
- what to do with it

If this is built into the beta script, it should be possible to add/remove that with a single instruction. (I maintain a similar large collection of data files that also have a public review phase and resorted to have the line %STATUS% in each source file and the release script changes that to "internal draft", "public review draft (with instructions)" or "" (empty string for the final release)). 

Some scheme like this seems indicated. (Could use that same method for internal or alpha drafts as well)

**Comment by markusicu** [comment]
We publish alpha and beta files in a location whose path contains "draft". This indicates the status at least as well as a version suffix. Some of the readmes that are next to the data files also say whether they are draft or final.

I have not heard from anyone that they were confused about the status of these files, and don't want to create unnecessary work.

I suspect that the previous presence of a version suffix was not giving much of a signal to UTC outsiders that that suffix meant "draft".

CLDR and ICU publish data files, but we don't go out of our way to switch their contents between "draft" and "release" all the time. And again, I am not aware of people being confused.

Some files are maintained manually.
Several files are generated by tools from Ken or Roozbeh etc., not these Unicode Tools here.

I suppose that we *could* add some kind of placeholder into each file which is replaced by the production script.
But it feels like a solution in search of a problem.

**Comment by asmusf** [comment]
If a colleague sends me a file, it doesn't help that it came from a "draft" folder. 

My browser opened it from a remembered location and I did not spot the "draft" in the address bar right now.

So that is not a professional way to handle this.

Finally, any copy I make to my machine, I will have to verify against the online to make sure I didn't accidentally grab a preliminary version.

Not being upfront about a version (or draft status) and hiding behind date stamps and folder locations is never good.

**Comment by asmusf** [comment]
From a discussion with @Ken-Whistler during the edcom meeting, I understand that you have some ability to supply copyright and similar information without actually editing all files. I'm simply suggesting to do something like that.

For example, in my other project, just by setting a pointer to a single file, I can insert the following notice into every one of over a hundred published files.

_This is a DRAFT document released for public comments and not final. Please see the announcement on the ... website for public comments on ... for details on how to submit comments._

Not only does any recipient know that this file is not final, but there's a succinct reminder of what to do with it. On publication that gets replaced with an empty string, and all is good.

If a subset of files aren't maintained using a workflow that makes that possible, that's not a reason to do nothing, in my view. Same if some files lack all comments for historical reasons.

**Comment by asmusf** [comment]
> My browser opened it from a remembered location \[or from some other link that was clearly did not contain "draft"\] and I did not spot the "draft" in the address bar \[when reading the file\] (happened to me right now).

(edits added)

At first, I thought this was just me, but then I realized that I had clicked on a link that did __**not**__ have a "draft" in it which I had found somewhat surprising. But then I read this in a recent pull request comment:

> The link checker complains about https://www.unicode.org/Public/UCA/16.0.0/ etc. because there should be redirects from there to /Public/draft/UCA/ but they seem to be missing. I sent ... an email about that.

I understand why such redirects are useful, but it means that we cannot (or definitely should not) rely on the live path to indicate the draft status of a file. For example, if someone jumps the gun and tries to access a data file with the final link, they will, before the end of the beta, get a beta file that (other than publication date) gives no indication it's not final. 

We can (and should) do better.

**Comment by markusicu** [comment]
> ... I read this in a recent pull request comment:
> 
> > The link checker complains about https://www.unicode.org/Public/UCA/16.0.0/ etc. because there should be redirects from there to /Public/draft/UCA/ but they seem to be missing. I sent ... an email about that.
> 
> I understand why such redirects are useful, but it means that we cannot (or definitely should not) rely solely on the live **_path_** to indicate the draft status of a file. For example, if someone jumps the gun and tries to access a data file with the final link, they will, before the end of the beta, get a beta file that (other than date stamp) gives no indication it's not final.

Note that the redirect is visible. The address bar will change to ".../Public/draft/...".

**Comment by asmusf** [comment]
> Note that the redirect is visible. The address bar will change to ".../Public/draft/...".

Yes, there's that subtle hint. But I argue that hiding behind a dynamically changed address bar is no way to run a standards organizations duty to be very explicit about what is a draft and what is final.

**Comment by markusicu** [comment]
Update: Each data file folder has a readme, and starting with the Unicode 18 beta, every readme says whether the files are draft or final.

I have not seen anyone else wishing for each data file to say so.

To do so would be expensive: While the readmes use a mini template system, the data files are generated by lots of different tools and don't get regenerated unless the data changes.

--> low ROI --> I propose that we close this issue.

**Comment by asmusf** [comment]
The BETA version of DerivedAge has a header that starts off this way:

```
# DerivedAge-18.0.0.txt
# Date: 2026-04-29, 19:43:55 GMT
# © 2026 Unicode®, Inc.
```

There's clearly some templating going on in the tool that creates it.

However, the template is indistinguishable from the PUBLISHED version (here for 17.0.0)

```
# DerivedAge-17.0.0.txt
# Date: 2025-07-30, 23:54:38 GMT
# © 2025 Unicode®, Inc.
```

That means, anytime one of these files is downloaded, you CANNOT tell whether you have a PUBLISHED or BETA file (unless you are an insider and can make educated guesses based on the date/timestamp). 

I don't think that is good. 

People download copies of these files (and we don't want them to read the live UCD folder for every access, so we effectively ask them to cache these files locally, or enter them into their own source control). I think we should not assume that people maintain our folder structure and therefore have the readme. Even then, you simply don't know whether you have some stale BETA file in your collection unless you parse and compare time stamps.

Depending on the tool, it should be possible to add the words "ALPHA", or "BETA" or simply "DRAFT" either after the time stamp, or after the file (or as part of the internal(!) file name before the extension).

**Comment by markusicu** [comment]
People generally have some context of where their data comes from.
If they have no idea where they come from, then they have no idea whether the files have been tampered with either.
In the absence of any real bug reports, this just seems like a theoretical concern and not worth spending limited time on.

**Comment by asmusf** [comment]
I see what you did here. 

The rule against cleaning up on theoretical grounds wasn't intended as an ROI metric but agreed to because of the very real potential for regressions in existing documents and implementations.

Let's keep it that way.

However, the limited resources argument stands on its own.

---
Incidentally, I'm not worried about "tampering" with files but with the reality of having worked on two different projects that needed a partial set of UCD files, including the need to preview by downloading and working with draft versions. Having succeeded in getting the draft info out of the file name helped immensely by being able to set up consumers with file names that do not have to change with every iteration.

But at the same time, it introduces the issue that it's harder to verify that a working set of UCD files isn't stuck on some late beta draft vs. the actual one. I've run into that issue, so you can consider that a "bug report", but it's of course possible to find some workaround.
